util/src/main/java/tech/pegasys/teku/util/config/KeyStoreFilesLocator.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.logging.log4j.Logger;
+
+public class KeyStoreFilesLocator {
+  private static final Logger LOG = LogManager.getLogger();
+  private final List<Pair<Path, Path>> filePairs;
+  private final Map<Path, Path> pathMap;
+  private final List<String> colonSeparatedPairs;
+  private final String pathSeparator;
+
+  public KeyStoreFilesLocator(
+      Optional<List<String>> maybeColonSeparatedPairs, final String pathSeparator) {
+    this.filePairs = new ArrayList<>();
+    this.colonSeparatedPairs = maybeColonSeparatedPairs.orElse(List.of());
+    this.pathMap = new HashMap<>();
+    this.pathSeparator = pathSeparator;
+  }
+
+  public void parse() {
+    for (final String currentEntry : colonSeparatedPairs) {
+      if (!currentEntry.contains(pathSeparator)) {
+        throw new InvalidConfigurationException(
+            "validatorKeys entry ("
+                + currentEntry
+                + ") did not contain key and password separated by '"
+                + pathSeparator
+                + "' as expected.");
+      }
+      if (currentEntry.matches(".*:.*:.*")) {
+        throw new InvalidConfigurationException(
+            "validatorKeys entry ("
+                + currentEntry
+                + ") contained more than one '"
+                + pathSeparator
+                + "', not keyFile'"
+                + pathSeparator
+                + "'passFile as expected.");
+      }
+      final List<String> entry = Splitter.on(pathSeparator).splitToList(currentEntry);
+      parseEntry(entry.get(0), entry.get(1));
+    }
+  }
+
+  public void parseKeyAndPasswordList(
+      final List<String> keystoreFiles, final List<String> keystorePasswordFiles) {
+    for (int i = 0; i < keystoreFiles.size(); i++) {
+      parseEntry(keystoreFiles.get(i), keystorePasswordFiles.get(i));
+    }
+  }
+
+  void parseEntry(final String keyFileName, final String passwordFileName) {
+    final File keyFile = new File(keyFileName);
+    final File passwordFile = new File(passwordFileName);
+
+    if (!keyFile.exists()) {
+      throw new InvalidConfigurationException(
+          String.format("Invalid configuration. could not find the key file (%s).", keyFileName));
+    }
+    if (!passwordFile.exists()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Invalid configuration. could not find the password file (%s).", passwordFileName));
+    }
+    if (keyFile.isDirectory() != passwordFile.isDirectory()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Invalid configuration. validatorKeys entry (%s"
+                  + pathSeparator
+                  + "%s) must be both directories or both files",
+              keyFileName,
+              passwordFileName));
+    }
+    if (keyFile.isFile()) {
+      pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
+      filePairs.add(Pair.of(keyFile.toPath(), passwordFile.toPath()));
```

Comment:
I'm not sure `filePairs` is doing anything useful, do we still need it?

Reply:
that was v.0 implementation floating around that i managed not to clean up :/ the map gives us easier de-duping, and i removed some but apparently not all of the old list.